### PR TITLE
Medical privacy

### PR DESCRIPTION
**Pull request by izwick-schachter**
As per #10, I began working on medical only statuses and medical only comments. Medical only comments are complete, and medical only statuses exists but have no view where an admin can check the logs of who changed status when. I have not worked on added a chart_code field yet, but it might be better to wait on #13 for that.

### CODE REVIEW

File: app/controllers/rescue_requests_controller.rb [hunk 1]
```diff
@@ -71,6 +72,15 @@ def apply_triage_status
     redirect_to disaster_request_path(disaster_id: @disaster.id, num: @request.incident_number)
   end
 
+  def apply_medical_triage_status
+    if current_user.has_role?(:medical) && @request.update(medical_status_id: params[:status_id])
```
Comment (ArtOfCode-) [inline app/controllers/rescue_requests_controller.rb]: Use a before_action callback to check roles, and use `has_any_role? :developer, :medical' in it

File: app/views/case_notes/_form.html.erb [hunk 2]
```diff
@@ -6,6 +6,13 @@
     <%= f.text_area :content, rows: 7, cols: 100, class: 'form-control' %>
   </div>
 
+  <% if current_user.has_role? :medical %>
```
Comment (ArtOfCode-) [inline app/views/case_notes/_form.html.erb]: Again, `current_user.has_any_role? :developer, :medical`

File: app/views/rescue_requests/show.html.erb [hunk 3]
```diff
@@ -77,7 +78,7 @@
 <h3>Case Notes <span class="text-small"><%= link_to 'Add', new_case_note_path(request_id: @request.id) %></span></h3>
 <% if @request.case_notes.any? %>
   <% @request.case_notes.each do |cn| %>
-    <%= render 'case_notes/case_note', case_note: cn, rid: @request.id %>
+    <%= render 'case_notes/case_note', case_note: cn, rid: @request.id unless cn.medical && !current_user.has_role?(:medical) %>
```
Comment (ArtOfCode-) [inline app/views/rescue_requests/show.html.erb]: Any role again

File: app/views/rescue_requests/triage_status.html.erb [hunk 4]
```diff
@@ -83,3 +84,15 @@
   <%= select_tag :status_id, options_for_select(@statuses.map { |s| [s.name, s.id] }), class: 'form-control form-control-sm mx-2' %>
   <%= submit_tag 'Save', class: 'btn btn-outline-success btn-sm mx-2' %>
 <% end %>
+
+<br>
+
+<% if current_user.has_role? :medical %>
```
Comment (ArtOfCode-) [inline app/views/rescue_requests/triage_status.html.erb]: And again

File: db/migrate/20170916194350_create_medical_statuses.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class CreateMedicalStatuses < ActiveRecord::Migration[5.1]
+  def change
+    create_table :medical_statuses do |t|
+      t.string :name
+      t.string :description
```
Comment (ArtOfCode-) [inline db/migrate/20170916194350_create_medical_statuses.rb]: You want a text type for description fields: common gotcha, strings are 255 chars max. You'll need to create a new migration with a change_column call.

**Comment by izwick-schachter** [comment]
@ArtOfCode- I addressed all of those issues.

File: app/controllers/rescue_requests_controller.rb [hunk 6]
```diff
@@ -117,6 +118,10 @@ def check_rescue
     require_any :developer, :admin, :rescue
   end
 
+  def check_medical
+    require_any :developera, :medical
```
Comment (ArtOfCode-) [inline app/controllers/rescue_requests_controller.rb]: Typo

File: app/views/case_notes/_form.html.erb [hunk 7]
```diff
@@ -6,7 +6,7 @@
     <%= f.text_area :content, rows: 7, cols: 100, class: 'form-control' %>
   </div>
 
-  <% if current_user.has_role? :medical %>
+  <% if current_user.has_role? :developer, :medical %>
```
Comment (ArtOfCode-) [inline app/views/case_notes/_form.html.erb]: <code>current_user.has_<em>any</em>_role?</code>

File: app/views/rescue_requests/show.html.erb [hunk 8]
```diff
@@ -78,7 +78,7 @@
 <h3>Case Notes <span class="text-small"><%= link_to 'Add', new_case_note_path(request_id: @request.id) %></span></h3>
 <% if @request.case_notes.any? %>
   <% @request.case_notes.each do |cn| %>
-    <%= render 'case_notes/case_note', case_note: cn, rid: @request.id unless cn.medical && !current_user.has_role?(:medical) %>
+    <%= render 'case_notes/case_note', case_note: cn, rid: @request.id unless cn.medical && !current_user.has_role?(:developer, :medical) %>
```
Comment (ArtOfCode-) [inline app/views/rescue_requests/show.html.erb]: `current_user.has_any_role?`

File: app/views/rescue_requests/triage_status.html.erb [hunk 9]
```diff
@@ -87,7 +87,7 @@
 
 <br>
 
-<% if current_user.has_role? :medical %>
+<% if current_user.has_role? :developer, :medical %>
```
Comment (ArtOfCode-) [inline app/views/rescue_requests/triage_status.html.erb]: `current_user.has_any_role?`

File: app/controllers/rescue_requests_controller.rb (hunk 6 above)
Comment (izwick-schachter) [reply app/controllers/rescue_requests_controller.rb]: Fixed

File: app/controllers/rescue_requests_controller.rb [hunk 10]
```diff
@@ -74,7 +74,9 @@ def apply_triage_status
   end
 
   def apply_medical_triage_status
+    prev_status = @request.medical_status&.name || "(no status)"
     if @request.update(medical_status_id: params[:status_id])
+      @request.case_notes.create(user_id: current_user.id, medical: true, content: "Changed status from #{prev_status} to #{@request.medical_status.name}")
```
Comment (ArtOfCode-) [inline app/controllers/rescue_requests_controller.rb]: If you create this with `user_id: nil` instead, it'll be attributed to System. You can mention the user in the text instead.

Comment (izwick-schachter) [reply app/controllers/rescue_requests_controller.rb]: Done

File: app/controllers/rescue_requests_controller.rb [hunk 11]
```diff
@@ -76,7 +76,7 @@ def apply_triage_status
   def apply_medical_triage_status
     prev_status = @request.medical_status&.name || "(no status)"
     if @request.update(medical_status_id: params[:status_id])
-      @request.case_notes.create(user_id: current_user.id, medical: true, content: "Changed status from #{prev_status} to #{@request.medical_status.name}")
+      @request.case_notes.create(user_id: nil, medical: true, content: "Changed status from #{prev_status} to #{@request.medical_status.name}")
```
Comment (ArtOfCode-) [inline app/controllers/rescue_requests_controller.rb]: You probably want to mention the user who made the change in the content of the note - "User 'ArtOfCode' (ID 7) changed status..." or something

Comment (izwick-schachter) [reply app/controllers/rescue_requests_controller.rb]: I just added username, those should be unique enough.

**Pull request merged**